packages/hydrogen/src/newsletter-subscribe/newsletter-subscribe.example.tsx

```diff
@@ -0,0 +1,51 @@
+import {UNSTABLE_NewsletterSubscribeForm as NewsletterSubscribeForm} from '@shopify/hydrogen';
+
+// place <NewsletterSignUp /> anywhere you want the subscription form to show up
+export function NewsletterSignUp({placeholder = 'Email'}) {
+  return (
+    <NewsletterSubscribeForm>
+      {(fetcher, isSuccessful, error) => {
+        return (
+          <>
+            <p>Subscribe to our emails</p>
+            <input
+              type="email"
+              name="email"
+              aria-required="true"
+              autoCorrect="off"
+              autoCapitalize="off"
+              autoComplete="email"
+              placeholder="Email"
+              required
+            />
+            {isSuccessful ? (
+              <span>✅</span>
+            ) : (
+              <button type="submit" disabled={fetcher.state !== 'idle'}>
+                Sign Up
+              </button>
+            )}
+            {error ? (
+              <p>
+                <em>{error}</em>
+              </p>
+            ) : null}
+          </>
+        );
+      }}
+    </NewsletterSubscribeForm>
+  );
+}
+
+// in app/routes/api.newsletter-subscribe.tsx by default
```

Comment:
Make this a 🪄 virtual route that combine the functionality of `storefrontRedirect` into a `shopifyRoutes` utility that uses the following order of operations:

//1. existing route files
//2. virtual ✨ resource routes
//3. storefront redirect